### PR TITLE
MCP-2664 POST exam-order-slack endpoint

### PR DESCRIPTION
**Pull request by ianwolson**
## What was the problem?
We want an endpoint to trigger a process that slacks all exam orders that have not yet been processed by MAS.

Associated tickets or Slack threads:

- [MCP-2664](https://amida.atlassian.net/browse/MCP-2664)

## How does this fix it?[^1]
Creates an exam-order-slack endpoint which will slack all exam orders that have not been processed by MAS.  aka orderedAt = null.

## How to test this PR
- Run automatedClaim with benefeitClaimId = 1390 and collectionId = 390
- Verify that the audit shows in DB, and that the mock-slack container contains the correct message.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java [hunk 1]
```diff
@@ -149,4 +149,37 @@ ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(
       @RequestParam(name = "size", required = false, defaultValue = "10")
           @Min(value = 1, message = "invalid size")
           Integer size);
+
+  @Operation(
+      summary = "Retrieves all exam order records",
```
Comment (hayleymdawson) [inline api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java:154]: Might want to mention slack in the summary just so people dont accidentally spam slack.

File: api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java [hunk 2]
```diff
@@ -149,4 +149,37 @@ ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(
...
+      description =
+          "This endpoint does the same as exam-order-info, but it also slacks the list of exam orders.")
```
Comment (hayleymdawson) [inline api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java:156]: Also unless it changed the other endpoint will return processed orders too and we don't want to slack those so maybe this needs to be more specific in the description?

**Review by hayleymdawson — CHANGES_REQUESTED** [review]
Code looks mostly clean, but one spec vs implementation difference- there's no filter on whether the status is ORDER_SUBMITTED or not (which is what VRO sets), that I see, which unless I missed it, are the only items requested? (or possibly some other status filter).

We wouldnt want to repeatedly slack things that are processed/finished.

**Review by au2 — CHANGES_REQUESTED** [review]
I would wait Julian's PR #1446 to finalize this.

File: service/provider/src/main/java/gov/va/vro/service/provider/services/ExamOrderSlackProcessor.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+public class ExamOrderSlackProcessor implements Processor {
+  @Override
+  public void process(Exchange exchange) {
+    ExamOrdersInfo exams = exchange.getIn().getBody(ExamOrdersInfo.class);
+    exchange
+        .getIn()
+        .setBody(AuditEvent.fromAuditable(exams, "exam-order-slack", getSlackMessage(exams)));
+  }
+
+  public static String getSlackMessage(ExamOrdersInfo exams) {
```
Comment (au2) [inline service/provider/src/main/java/gov/va/vro/service/provider/services/ExamOrderSlackProcessor.java:21]: why is this public?

File: api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java [hunk 4]
```diff
@@ -149,4 +149,37 @@ ResponseEntity<List<ExamOrderInfoResponse>> allExamOrderInfo(
...
+  @Tag(name = "Claim Metrics")
+  @RequestMapping(value = "/exam-order-slack", method = RequestMethod.POST)
+  @ResponseBody
+  ResponseEntity<List<ExamOrderInfoResponse>> examOrderSlack(
+      @RequestParam(name = "page", required = false, defaultValue = "0")
+          @Min(value = 0, message = "invalid page number")
+          Integer page,
+      @RequestParam(name = "size", required = false, defaultValue = "10")
+          @Min(value = 1, message = "invalid size")
+          Integer size);
```
Comment (au2) [inline api/src/main/java/gov/va/vro/api/resources/ClaimMetricsResource.java:184]: Add the param added in PR 1664

File: service/provider/src/main/java/gov/va/vro/service/provider/services/ExamOrderSlackProcessor.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import gov.va.vro.model.claimmetrics.ExamOrdersInfo;
+import gov.va.vro.model.claimmetrics.response.ExamOrderInfoResponse;
+import gov.va.vro.model.event.AuditEvent;
+import gov.va.vro.service.provider.mas.MasException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+@Slf4j
+public class ExamOrderSlackProcessor implements Processor {
```
Comment (au2) [inline service/provider/src/main/java/gov/va/vro/service/provider/services/ExamOrderSlackProcessor.java:12]: Do we really need this processor? Why don't you prepare the body before entering to the camel route and just enter with that one?

**Review by au2 — CHANGES_REQUESTED** [review]
Some inline comments. Please address the conflicts. I have a feeling that that would want claimid in the body but that would be another ticket.

**Comment by ianwolson** [comment]
Refactor and conflict resolution completely broke this :/  will continue trying to fix this.

**Comment by ianwolson** [comment]
Nevermind, starting fresh on a new branch.  This one is hopeless.

**Pull request closed without merging**